yt_dlp/extractor/bitchute.py

```diff
@@ -69,82 +69,101 @@ def _real_extract(self, url):
+            'id': 'wV9Imujxasw9',
+            'title': 'Bruce MacDonald and "The Light of Darkness"',
+            'description': 'md5:04913227d2714af1d36d804aa2ab6b1e',
+        }
+    }]
 
     _TOKEN = 'zyG6tQcGPE5swyAEFLqKUwMuMMuF6IO2DZ6ZDQjGfsL0e4dcTLwqkTTul05Jdve7'
+    PAGE_SIZE = 25
+    HTML_CLASS_NAMES = {
+        'channel': {
+            'container': 'channel-videos-container',
+            'title': 'channel-videos-title',
+            'description': 'channel-videos-text',
+        },
+        'playlist': {
+            'container': 'playlist-video',
+            'title': 'title',
+            'description': 'description',
+        }
+
+    }
 
-    def _entries(self, channel_id):
-        channel_url = 'https://www.bitchute.com/channel/%s/' % channel_id
-        offset = 0
-        for page_num in itertools.count(1):
+    def _entries(self, playlist_type, playlist_id):
+        playlist_url = 'https://www.bitchute.com/%s/%s/' % (playlist_type, playlist_id)
+
+        def fetch_entries(page_num):
             data = self._download_json(
-                '%sextend/' % channel_url, channel_id,
-                'Downloading channel page %d' % page_num,
+                '%sextend/' % playlist_url, playlist_id,
+                'Downloading %s page %d' % (playlist_type, page_num),
                 data=urlencode_postdata({
                     'csrfmiddlewaretoken': self._TOKEN,
                     'name': '',
-                    'offset': offset,
+                    'offset': page_num * self.PAGE_SIZE,
                 }), headers={
                     'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
-                    'Referer': channel_url,
+                    'Referer': playlist_url,
                     'X-Requested-With': 'XMLHttpRequest',
                     'Cookie': 'csrftoken=%s' % self._TOKEN,
                 })
-            if data.get('success') is False:
-                break
-            html = data.get('html')
-            if not html:
-                break
-            video_ids = re.findall(
-                r'class=["\']channel-videos-image-container[^>]+>\s*<a\b[^>]+\bhref=["\']/video/([^"\'/]+)',
-                html)
-            if not video_ids:
-                break
-            offset += len(video_ids)
-            for video_id in video_ids:
+            if data.get('success') is not True:
+                return
+            class_name = self.HTML_CLASS_NAMES[playlist_type]
+            for video_html in get_elements_html_by_class(class_name['container'], data.get('html')):
+                match = re.search(r'<a\b[^>]+\bhref=["\']/video/(?P<id>[^"\'/]+)', video_html)
+                video_id = match and match.group('id')
                 yield self.url_result(
                     'https://www.bitchute.com/video/%s' % video_id,
-                    ie=BitChuteIE.ie_key(), video_id=video_id)
+                    ie=BitChuteIE, video_id=video_id, url_transparent=True,
```

Comment:
Why transparent?

Reply:
Because `duration` and `view_count` cannot be extracted by `BitChuteIE`, so they will be propagated.
https://github.com/yt-dlp/yt-dlp/blob/1305b659ef2bf3c76851b9400c7ac4a8f100fce2/yt_dlp/YoutubeDL.py#L1678-L1680

Does this have any other implications?